oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/InitiateIPVAuthorisationService.java

```diff
@@ -127,9 +133,27 @@ public APIGatewayProxyResponseEvent sendRequestToIPV(
                 null);
     }
 
-    private Optional<ClaimsSetRequest> buildIpvClaimsRequest(AuthenticationRequest authRequest) {
-        return Optional.ofNullable(authRequest)
-                .map(AuthenticationRequest::getOIDCClaims)
-                .map(OIDCClaimsRequest::getUserInfoClaimsRequest);
+    private Optional<ClaimsSetRequest> buildIpvClaimsRequest(
+            AuthenticationRequest authRequest, Subject internalPairwiseSubject) {
+
+        Optional<ClaimsSetRequest> claimsSetRequest =
+                Optional.ofNullable(authRequest)
+                        .map(AuthenticationRequest::getOIDCClaims)
+                        .map(OIDCClaimsRequest::getUserInfoClaimsRequest);
+
+        if (configurationService.sendStorageTokenToIpvEnabled()) {
+            LOG.info("Adding storageAccessToken claim to IPV claims request");
+            AccessToken storageToken =
+                    storageTokenService.generateAndSignStorageToken(
+                            internalPairwiseSubject, JWSAlgorithm.ES256);
+            claimsSetRequest.ifPresent(
+                    csr ->
+                            csr.add(
+                                    new ClaimsSetRequest.Entry(
+                                                    configurationService.getStorageTokenClaimName())
+                                            .withValues(List.of(storageToken.getValue()))));
+        }
```

Comment:
Not sure this is quite right? At the moment is the claimsSetRequest is null that's probably fine. It's a bit weird to ask someone to prove their identity and not get it back, but it's possible. After this change, it would no longer be possible as if there were no claims we would not set the storageAccessToken